CleverGirl/CleverGirl/Objects/CandidateWeapons.cs

```diff
@@ -69,37 +67,44 @@ public CandidateWeapons(AbstractActor attacker, ICombatant target)
                 }
 
                 // WillFireAtTargetFromPosition has an implicit check in CAC for minimum range. False can mean there's a possible shot, but weapon mode is limiting the action
-                bool willFireAtTarget = rawWeapon.WillFireAtTargetFromPosition(target, attacker.CurrentPosition, attacker.CurrentRotation);
-                bool withinRange = distance <= rawWeapon.MaxRange;
-                Mod.Log.Debug?.Write($" -- base weaponAndAmmo has willFire: {willFireAtTarget}  willFire: {withinRange}");
-
-                bool canAttack = willFireAtTarget && withinRange;
-
-                // Iterate weapon and ammo modes
+                bool canFireAtTarget = false;
+                bool hasLOF = false;
                 List<AmmoModePair> firingMethods = rawWeapon.getAvaibleFiringMethods();
                 if (firingMethods != null && firingMethods.Count > 0)
                 {
-                    foreach (AmmoModePair item in rawWeapon.getAvaibleFiringMethods())
+                    AmmoModePair currentAmmoMode = rawWeapon.getCurrentAmmoMode();
+                    foreach (AmmoModePair ammoModePair in rawWeapon.getAvaibleFiringMethods())
                     {
-                        rawWeapon.ApplyAmmoMode(item);
+                        rawWeapon.ApplyAmmoMode(ammoModePair);
                         bool modeWillFire = rawWeapon.WillFireAtTargetFromPosition(target, attacker.CurrentPosition, attacker.CurrentRotation);
```

Comment:
IIRC, using attacker.CurrentPosition is a bug in current CG. Since you're changing the surrounding logic, you might want to change this to the attack position instead.

Reply:
Not sure what attack position would be otherwise, we don't know any movement here do we? All the logic uses CurrentPosition as far as I can find.